Add configuration information for paypload_reset

ShadowJonathan opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

This adds the necessary configuration documentation for MQTT Device Tracker auto-zoning.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/83952

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

ShadowJonathan committed:
add configuration information for paypload_auto_zone

jbouwh commented:
Please also update the abbreviations list for Mqtt discovery, found at the integration page.

ShadowJonathan committed:
abbr

ShadowJonathan committed:
change abbreviation

frenck commented on source/_integrations/device_tracker.mqtt.markdown:
```diff
@@ -74,6 +74,11 @@ payload_not_home:
   type: string
   default: "not_home"
+payload_auto_zone:
+  description: The payload value that will have the device's location automatically derived from home assistant's zones.
```
```suggestion
  description: The payload value that will have the device's location automatically derived from Home Assistant's zones.
```

frenck commented on source/_integrations/device_tracker.mqtt.markdown:
```diff
@@ -205,6 +210,11 @@ payload_not_home:
   type: string
   default: not_home
+payload_auto_zone:
+  description: The payload value that will have the device's location automatically derived from home assistant's zones.
```
```suggestion
  description: The payload value that will have the device's location automatically derived from Home Assistant's zones.
```

frenck committed:
Tiny tweak

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.

ShadowJonathan committed:
change auto_zone to reset

ShadowJonathan commented:
Update: I rebased on `next` and changed `auto_zone` to `reset` as per [this review](https://github.com/home-assistant/core/pull/83952#discussion_r1051980416)

ShadowJonathan committed:
change RST to None, address isort
